modules/navier_stokes/src/fvkernels/NSFVFrictionFlowDiode.C

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NSFVFrictionFlowDiode.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", NSFVFrictionFlowDiode);
+
+InputParameters
+NSFVFrictionFlowDiode::validParams()
+{
+  InputParameters params = FVElementalKernel::validParams();
+  params.addClassDescription("Adds a linear friction term, -K |v*d|, "
+                             "if the flow is opposite the direction d of the diode");
+  params.addRequiredParam<Real>("resistance",
+                                "Friction factor multiplying the superficial velocity if the flow "
+                                "is in a half plane in the opposite direction of the normal");
+  params.addRequiredParam<RealVectorValue>(
+      "direction",
+      "Normal direction of the diode. Flow is free in this half-plane, "
+      "subject to friction in the other halfplane");
+
+  // Defaults are based on porous media as this is a more likely use case
+  params.addParam<MooseFunctorName>(
+      "u", NS::superficial_velocity_x, "X-(superficial) velocity functor");
+  params.addParam<MooseFunctorName>(
+      "v", NS::superficial_velocity_y, "Y-(superficial) velocity functor");
+  params.addParam<MooseFunctorName>(
+      "w", NS::superficial_velocity_z, "Z-(superficial) velocity functor");
+
+  return params;
+}
+
+NSFVFrictionFlowDiode::NSFVFrictionFlowDiode(const InputParameters & params)
+  : FVElementalKernel(params),
+    _direction(getParam<RealVectorValue>("direction")),
+    _resistance(getParam<Real>("resistance")),
+    _u(getFunctor<ADReal>("u")),
+    _v(getFunctor<ADReal>("v")),
+    _w(getFunctor<ADReal>("w"))
+{
+}
+
+ADReal
+NSFVFrictionFlowDiode::computeQpResidual()
+{
+  const auto elem_arg = makeElemArg(_current_elem);
+  const ADRealVectorValue elem_velocity(_u(elem_arg), _v(elem_arg), _w(elem_arg));
+
+  if (elem_velocity * _direction < 0)
+    return _resistance * std::abs(elem_velocity * _direction);
+  else
+    return 0;
```

Comment:
```suggestion
  const auto elem_arg = makeElemArg(_current_elem);
  const auto vel_comp = _var(elem_arg);

  if (vel_comp * _direction(_index) < 0)
    return _resistance * vel_comp * std::abs(_direction(_index));
  else
    return 0;
```
I think it should be something like this or else you are adding body forces in transverse flow directions?

Reply:
It's a good point. In my mind I wanted to penalize every component by the same amount K. Then I decided K |v| would be better behaved. But I guess K v_i is the more fair one